eos-migrate-chromium-profile

```diff
@@ -0,0 +1,150 @@
+    is a handler for, because we can't construct a GDesktopAppInfo for it, because its
+    desktop file no longer exists.
+    """
+
+    keyfile = GLib.KeyFile()
+    try:
+        keyfile.load_from_file(
+            path, GLib.KeyFileFlags.KEEP_COMMENTS | GLib.KeyFileFlags.KEEP_TRANSLATIONS,
+        )
+    except GLib.GError as gerror:
+        if gerror.matches(GLib.file_error_quark(), GLib.FileError.NOENT):
+            return
+
+        raise
+
+    changed = False
+
+    for group in MIMEAPPS_GROUPS:
+        if not keyfile.has_group(group):
+            continue
+
+        keys, _length = keyfile.get_keys(group)
+        for key in keys:
+            values = keyfile.get_string_list(group, key)
+            try:
+                i = values.index(OLD_DESKTOP_FILE)
+            except ValueError:
+                pass
+            else:
+                values[i] = NEW_DESKTOP_FILE
+                keyfile.set_string_list(group, key, values)
+                changed = True
+
+    if changed:
+        keyfile.save_to_file(path)
+
+
+def update_desktop_shortcut(path):
+    keyfile = GLib.KeyFile()
+    print("path: " + path)
+    keyfile.load_from_file(path, GLib.KeyFileFlags.NONE)
+
+    group = "Desktop Entry"
+    if not keyfile.has_group(group):
+        return
+
+    exec_cmd = keyfile.get_string(group, "Exec")
+    if not exec_cmd.startswith("/usr/bin/chromium-browser"):
+        return
+
+    keyfile.set_string(group, "Exec",
+                       "/usr/bin/flatpak run org.chromium.Chromium" +
+                       exec_cmd[len("/usr/bin/chromium-browser"):])
+    keyfile.set_string(group, "X-Flatpak-Part-Of", "org.chromium.Chromium")
+    keyfile.save_to_file(path)
+
+
+def update_desktop_shortcuts():
+    for filename in os.listdir(DESKTOP_SHORTCUTS_DIR):
+        if filename.endswith(".desktop"):
+            update_desktop_shortcut(os.path.join(DESKTOP_SHORTCUTS_DIR,
+                                                 filename))
+
+
+def update_old_config_references(path):
+    if not os.path.isfile(path):
+        return
+
+    for line in fileinput.input(path, inplace=True):
+        line = line.replace(OLD_CHROMIUM_CONFIG_DIR, NEW_CHROMIUM_CONFIG_DIR)
+        line = line.replace(OLD_WIDEVINE_SYSTEM_DIR, "")
+        print(line, end='')
+
+
+def main():
+    migrated_file = os.path.join(NEW_CHROMIUM_DATA_DIR, ".migrated")
+    if os.path.exists(migrated_file):
+        return
+
+    update_mimeapps_list(MIMEAPPS_LIST)
+    update_desktop_shortcuts()
+
+    if (
+        os.path.isdir(OLD_CHROMIUM_CONFIG_DIR) and
+        not os.path.isdir(NEW_CHROMIUM_CONFIG_DIR)
+    ):
+        os.makedirs(NEW_CHROMIUM_CONFIG_DIR)
+        os.rename(OLD_CHROMIUM_CONFIG_DIR, NEW_CHROMIUM_CONFIG_DIR)
+        update_old_config_references(
+            os.path.join(NEW_CHROMIUM_CONFIG_DIR,
+                         "WidevineCdm", "latest-component-updated-widevine-cdm"))
```

Comment:
If I were feeling pedantic I'd say: well, this should run before the rename, because otherwise if the script gets interrupted between the `rename` and this block, then gets re-run on the next boot, the Widevine stuff won't update.

I think it's unlikely.

Reply:
Done.